### PR TITLE
Catalyst registration APDU docs

### LINKED ISSUE #67: Add docs for Catalyst APDU flow
In the Catalyst PR (https://github.com/vacuumlabs/ledger-app-cardano-shelley/pull/65) we forgot to update the docs for the APDUs

### PR DESCRIPTION
**Pull request by refi93**
Motivation: add missing docs for Catalyst registration

Changes:
* add docs for Catalyst registration to a dedicated file
* fix numbering of steps in ins_sign_tx.md
* fix expected length of signatures in ins_sign_tx.md
* rename `AUX_DATA_TYPE_CATALYST_VOTING_KEY_REGISTRATION` to `AUX_DATA_TYPE_CATALYST_REGISTRATION` as the original was awkwardly long and inconsistent with the rest of the codebase (leftover from refactoring)

closes #67

### CODE REVIEW

File: doc/ins_sign_catalyst_registration.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Catalyst Voting Key Registration
+
+## Description
+
+Cardano uses a sidechain for its treasury system. One needs to "register" to participate on this sidechain by submitting a registration transaction on the mainnet chain. This CIP details the registration transaction format.
```
Comment (janmazak) [inline doc/ins_sign_catalyst_registration.md:5]: "This CIP" sounds out of place.

File: doc/ins_sign_catalyst_registration.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+A Catalyst registration transaction are a regular Cardano transaction with a specific transaction metadata associated with it.
```
Comment (janmazak) [inline doc/ins_sign_catalyst_registration.md:7]: I would keep it more technical here, the metadata are actually part of the tx, not just somehow "associated". And perhaps mention that the metadata is independent of tx body.

File: doc/ins_sign_catalyst_registration.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+In the following list of APDU messages (which are to be sent in the listed order), we only give the value of P2 and the format of message data. The general format of the message is
+
+|Field|Value|
+|-----|-----|
+| CLA | `0xD7` |
+| INS | `0x21` |
+|  P1 | `0x08` |
+|  P2 | (specific for each subcall) |
+
+All the resposes are empty.
```
Comment (janmazak) [inline doc/ins_sign_catalyst_registration.md:22]: One is not.

File: doc/ins_sign_catalyst_registration.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+**Nonce**
+
+P2 = `0x33`
+
+*Data*
+
+|Field| Length | Comments|
+|-----|--------|--------|
+|Nonce| 8| Big endian|
```
Comment (janmazak) [inline doc/ins_sign_catalyst_registration.md:72]: Hm. Don't we specify if it is int or uint? Apparently not, but I guess we should, everywhere. (Int is rare, will be used in for multiasset in minting. I guess everything until now was uint.) Perhaps create an issue to fix it later?

File: doc/ins_sign_catalyst_registration.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+---
+
+**Confirmation**
+
+P2 = `0x34`
+
+Data must be empty, however
```
Comment (janmazak) [inline doc/ins_sign_catalyst_registration.md:80]:
```suggestion
Data must be empty.
```

File: doc/ins_sign_tx.md [hunk 6]
```diff
@@ -82,7 +108,7 @@ Initializes signing request.
 |output index |  4 | Big endian |
 
 
-### 3 - Set outputs
+### 4 - Set outputs
```
Comment (janmazak) [inline doc/ins_sign_tx.md:111]: Is there any point in keeping these numbers? Let's just delete them, they do not correspond to tx body keys anymore anyway.

File: doc/ins_sign_tx.md [hunk 7]
```diff
@@ -64,7 +64,33 @@ Initializes signing request.
...
+|  P2 | (unused / see [Catalyst Registration](ins_sign_catalyst_registration.md)) |
+
+**Data for AUX_DATA_TYPE_ARBITRARY_HASH**
+
+Ledger cannot parse and display generic auxiliary data in full because their structure is too loose and their memory footprint potentially too big.
+So only the hash is transferred and displayed and the user has to use other means of verification that the hash is correct.
+
+|Field| Length | Comments|
+|-----|--------|--------|
+| Auxiliary data type | 1 | |
```
Comment (janmazak) [inline doc/ins_sign_tx.md:83]: I think we should mention the value in the last column (cf. `CERTIFICATE_TYPE_STAKE_REGISTRATION=0x00` elsewhere). Also below.

**Review by janmazak — APPROVED** [review]
I don't need to see this again, so I approved it.

File: doc/ins_sign_catalyst_registration.md (hunk 1 above)
Comment (refi93) [reply doc/ins_sign_catalyst_registration.md:5]: yeah, I just copied the first paragraph from the CIP, thanks for noticing

File: doc/ins_sign_catalyst_registration.md (hunk 4 above)
Comment (refi93) [reply doc/ins_sign_catalyst_registration.md:72]: fair point, I made an issue for it: https://github.com/vacuumlabs/ledger-app-cardano-shelley/issues/69

**Comment by refi93** [comment]
made PR to upstream repo: https://github.com/LedgerHQ/app-cardano/pull/9

**Pull request merged**